runtime/parsing.c

```diff
@@ -166,13 +170,13 @@ CAMLprim value caml_parse_engine(struct parser_tables *tables,
   case TOKEN_READ:
     RESTORE;
     if (Is_block(arg)) {
-      env->curr_char = Field(tables->transl_block, Tag_val(arg));
+      env->curr_char = Val_int(Int_field(tables->transl_block, Tag_val(arg)));
```

Comment:
`Val_int(Int_field` is a bit of a pain.  There is no longer a read barrier, so I'd rather go back to the original code, using just `Field`.  Same comment three lines below.